static/js/index.js

```diff
@@ -0,0 +1,147 @@
+    const idsModeNode = document.querySelectorAll('input[name="ids-radio-group"]')
+    loopModeNode.forEach( (el) => {
+        if (el.value === scraperSettings['loop_mode'].toString()) {
+            el.checked = true
+        }
+    })
+
+    mediaModeNode.forEach((el) => {
+        if (el.value === scraperSettings['media_mode'].toString()) {
+            el.checked = true
+        }
+    })
+
+    idsModeNode.forEach((el) => {
+        if (el.value === scraperSettings['ids_source'].toString()) {
+            el.checked = true
+        }
+    })
+}
+
+const checkShutdown = async () => {
+    const timeInterval = 2000
+    const makeRequest = async () => {
+        const requestUrl = `${baseUrl}/${API_PREFIX}/status/`;
+        const responseData = await (await fetch(requestUrl)).json()
+        return responseData
+    }
+
+    const updatePage = async () => {
+        try {
+            const response = await makeRequest()
+            if (response['status'] === "stopped") {
+                location.reload()
+            } else {
+                setTimeout(updatePage, timeInterval)
+            }
+         } catch {
+             location.reload()
+         }
+    }
+
+    setTimeout(updatePage, timeInterval)
+}
+
+const updateCommandButton = (buttonStatus) => {
+    const commandButton = document.getElementById('command-button')
+    if (buttonStatus === "running") {
+        commandButton.disabled = false
+        commandButton.classList = "btn stop-btn"
+        commandButton.innerText = "STOP"
+        commandButton.value = "stop"
+    } else if (buttonStatus === "shutdown") {
+        commandButton.disabled = true
+        commandButton.classList = "btn update-btn"
+        commandButton.innerText = "updating..."
+        checkShutdown()
+    } else {
+        commandButton.disabled = false
+        commandButton.classList = "btn start-btn"
+        commandButton.innerText = "START"
+        commandButton.value = "start"
+    }
+}
+
+const getScraperStatus = async () => {
+    const requestUrl = `${baseUrl}/${API_PREFIX}/index/`;
+    const responseData =  await (await fetch(requestUrl)).json()
+    const outputNode = document.getElementById('scraping-results')
+    setScrapingLogs(responseData['log_lines'])
+    setRadioButtons(responseData['scraper_settings'])
+    updateCommandButton(responseData['scraper_status'])
+    outputNode.innerText = `${responseData['output']['scraped_media'] ?? 0} media scraped`
+}
+
+const startScraping = async () => {
+    const userLimit = Number(document.getElementById('user-limit').value)
+    const loopMode = document.querySelector('input[name="loop-radio-group"]:checked').value
+    const mediaMode = document.querySelector('input[name="media-radio-group"]:checked').value
+    const idsMode = document.querySelector('input[name="ids-radio-group"]:checked').value
+    const commandType = document.getElementById('command-button').value
+    const errorField = document.getElementById('errors')
```

Comment:
Move this below, you don't need to query this if the user is logged in.

Reply:
Done